tests/Fixer/ControlStructure/SimplifiedIfReturnFixerTest.php

```diff
@@ -0,0 +1,250 @@
+            ],
+            'bug1' => [
+                '<?php if ($bar) { return 1; } return (bool) ($foo)      ;',
+                '<?php if ($bar) { return 1; } if ($foo) { return true; } return false;',
+            ],
+            'bug1-negative' => [
+                '<?php if ($bar) { return 1; } return ! ($foo)      ;',
+                '<?php if ($bar) { return 1; } if ($foo) { return false; } return true;',
+            ],
+            'bug1-braceless' => [
+                '<?php if ($bar) return 1; return (bool) ($foo)    ;',
+                '<?php if ($bar) return 1; if ($foo) return true; return false;',
+            ],
+            'bug1-braceless-negative' => [
+                '<?php if ($bar) return 1; return ! ($foo)    ;',
+                '<?php if ($bar) return 1; if ($foo) return false; return true;',
+            ],
+            [
+                <<<'EOT'
+<?php
+function f1() { return (bool) ($f1)      ; }
+function f2() { return true; } return false;
+function f3() { return (bool) ($f3)      ; }
+function f4() { return true; } return false;
+function f5() { return (bool) ($f5)      ; }
+function f6() { return false; } return true;
+function f7() { return ! ($f7)      ; }
+function f8() { return false; } return true;
+function f9() { return ! ($f9)      ; }
+EOT
+                ,
+                <<<'EOT'
+<?php
+function f1() { if ($f1) { return true; } return false; }
+function f2() { return true; } return false;
+function f3() { if ($f3) { return true; } return false; }
+function f4() { return true; } return false;
+function f5() { if ($f5) { return true; } return false; }
+function f6() { return false; } return true;
+function f7() { if ($f7) { return false; } return true; }
+function f8() { return false; } return true;
+function f9() { if ($f9) { return false; } return true; }
+EOT
+                ,
+            ],
+            'preserve-comments' => [
+                <<<'EOT'
+<?php
+// C1
+return (bool)
+# C2
+(
+/* C3 */
+$foo
+/** C4 */
+)
+// C5
+
+# C6
+
+// C7
+
+# C8
+
+/* C9 */
+
+/** C10 */
+
+// C11
+
+# C12
+;
+/* C13 */
+EOT
+                ,
+                <<<'EOT'
+<?php
+// C1
+if
+# C2
+(
+/* C3 */
+$foo
+/** C4 */
+)
+// C5
+{
+# C6
+return
+// C7
+true
+# C8
+;
+/* C9 */
+}
+/** C10 */
+return
+// C11
+false
+# C12
+;
+/* C13 */
+EOT
+                ,
+            ],
+            'preserve-comments-braceless' => [
+                <<<'EOT'
+<?php
+// C1
+return (bool)
+# C2
+(
+/* C3 */
+$foo
+/** C4 */
+)
+// C5
+# C6
+
+// C7
+
+# C8
+
+/* C9 */
+/** C10 */
+
+// C11
+
+# C12
+;
+/* C13 */
+EOT
+                ,
+                <<<'EOT'
+<?php
+// C1
+if
+# C2
+(
+/* C3 */
+$foo
+/** C4 */
+)
+// C5
+# C6
+return
+// C7
+true
+# C8
+;
+/* C9 */
+/** C10 */
+return
+// C11
+false
+# C12
+;
+/* C13 */
+EOT
+                ,
+            ],
+            'else-if' => [
+                '<?php if ($bar) { return $bar; } else return (bool) ($foo)      ;',
```

Comment:
`else` can be dropped (as done with `elseif`).

Reply:
If this context, `elseif` and `else if` are slightly different: `elseif` is findable and handable like `if`; `else if` instead can also be present as `else { if` and therefor be more complex to handle correctly. Since integration with `NoUselessElse` is already granted, I don't see the perks of adding complexity to the fixer in order to handle this case.

Reply:
I understand, [maybe later](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3253#discussion_r352009528) then.